queries.sql

```diff
@@ -125,3 +125,110 @@ ON owner_id = owners.id
 GROUP BY full_name
 ORDER BY COUNT(owner_id) DESC
 LIMIT 1;
+
+--Who was the last animal seen by William Tatcher?
+SELECT vets.name AS vet_name,
+  animals.name AS animal_name,
+  visit_date AS last_visit
+FROM vets
+JOIN visits ON vets.id = vet_id
+JOIN animals ON animal_id = animals.id
+WHERE vets.name = 'William Tatcher'
+GROUP BY vets.name, animals.name, visit_date
+ORDER BY MAX(visit_date) DESC
+LIMIT 1;
+
+--How many different animals did Stephanie Mendez see?
+SELECT vet_name,
+SUM(count) AS number_of_animals
+FROM (
+  SELECT vets.name AS vet_name,
+  COUNT(animal_id) AS count
+  FROM vets
+  JOIN visits ON vet_id = vets.id
+  WHERE vets.name = 'Stephanie Mendez'
+  GROUP BY vets.name, animal_id
+) AS subquery
+GROUP BY vet_name;
+
+--List all vets and their specialties, including vets with no specialties.
+SELECT
+  vets.name AS vet_name,
+  species.name AS specialty
+FROM vets
+LEFT JOIN specializations ON vets.id = vet_id
+LEFT JOIN species ON specialization_id = species.id;
+
+--List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT
+  animals.name AS animal,
+  vets.name AS vet_name,
+  visit_date
+FROM animals
+JOIN visits ON animal_id = animals.id
+JOIN vets ON vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez'
+AND visit_date > '2020-04-01' and visit_date < '2020-08-30';
+
+--What animal has the most visits to vets?
+SELECT
+  animals.name AS animal,
+  COUNT(visits.animal_id) AS number_of_visits
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+GROUP BY animals.name
+ORDER BY COUNT(visits.animal_id) DESC
+LIMIT 1;
```

Comment:
As per the screenshot you have attached for this query, it is returning the wrong results. The result of this query is returning `Squirtle` which is in fact the least visiting animal. Kindly fix this issue.